mongo/database_test.go

```diff
@@ -94,6 +95,43 @@ func TestDatabase(t *testing.T) {
 		_, err = db.ListCollections(bgCtx, bson.D{})
 		assert.Equal(t, ErrClientDisconnected, err, "expected error %v, got %v", ErrClientDisconnected, err)
 	})
+	t.Run("TransientTransactionError label", func(t *testing.T) {
+		client := setupClient(options.Client().ApplyURI("mongodb://nonexistent").SetServerSelectionTimeout(3 * time.Second))
+		err := client.Connect(bgCtx)
+		defer client.Disconnect(bgCtx)
+		assert.Nil(t, err, "expected nil, got %v", err)
+
+		t.Run("negative case of non-transaction", func(t *testing.T) {
+			var sse topology.ServerSelectionError
+			var le LabeledError
+
+			err := client.Ping(bgCtx, nil)
+			assert.NotNil(t, err, "expected error, got nil")
+			assert.True(t, errors.As(err, &sse), `expected error to be a "topology.ServerSelectionError"`)
+			if errors.As(err, &le) {
```

Comment:
Optional: If we wrap all errors returned by `selectServer` in a `driver.Error`, we can expect the error here to always be a `LabeledError`, which would make this test assertion more robust.